src/test/java/org/assertj/core/perf/ContainsOnlyPerfTest.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2019 the original author or authors.
+ */
+package org.assertj.core.perf;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.concurrent.ThreadLocalRandom;
+
+import org.junit.Test;
+
+/**
+ * See https://github.com/joel-costigliola/assertj-core/issues/1718
+ */
+public class ContainsOnlyPerfTest {
```

Comment:
Unfortunately this can't be an automated unit test, I don't want this test to fail only because it was run on a slow machine.

we can keep that class though but without the test part, just a good old main method (or simply disable the tests)
if we keep the tests then I would prefer them to be written with JUnit 5.

Reply:
To fail on a slow machine, each operation should take ~1 microsecond = 1000 nanoseconds, whereas the "operation" is some cheap manipulation with Array, with memory which resides caches. I don't believe there is such a test agent where that would fail, perhaps apart from some ultra-micro-embedded machine, in conjunction with JVM running in interactive (no compilation) mode.

OTOH, O(N^2) complexity pushes the time of this test far beyond 1 second, on a however powerful machine. Therefore I think this is a good guard against accidental regression to an O(N^2) complexity.

I could change this timeout to 10 seconds, which would still absolutely surely fail on any machine if the complexity was O(N^2).

Reply:
happy with 10s + adding a comment explaining that.